merger_sort.c

```diff
@@ -0,0 +1,98 @@
+#include <stdio.h>
+
+#define SIZE 8
+
+int temp[SIZE] = {0};
```

Comment:
I'm not sure a global array is the best idea, especially since this is going to be constant size. What you probably want is a local temporary array/arrays within merge function.